Adapted to allow 3D tracking along side 2D

pet1330 opened the pull request:
Adapted to allow 3D tracking along size 2D. I've tested the 3D tracking works. The 2D should not be affected by this, however, I've not had a chance to test this yet.

cdondrup commented:
Briefly tested on robot. Should be fine. @marc-hanheide @nbellotto any objections?

marc-hanheide commented on examples/simple_tracking.cpp:
```diff
@@ -112,6 +112,7 @@ int main(int argc, char *argv[]) {
   MultiTracker<Filter, 4> mtrk;    // state [x, v_x, y, v_y]
   CVModel cvm(5.0, 5.0);           // CV model with sigma_x = sigma_y = 5.0
   CartesianModel ctm(0.5, 0.5);              // Cartesian observation model
+  CartesianModel3D ctm(0.5, 0.5);              // Cartesian observation model
```
This is not ideal, maybe put a comment that people need to decide which one to instantiate.

cdondrup replied:
Right, shouldn't the 3D model have 3 arguments anyway? The example is, by default, not compiled otherwise this would have crashed I guess.
I'd say remove that line and keep the example purely 2D. It would make more sense to copy this file and make a separate 3D example because a few other things would have to be changed for this as well. But for now, as I said, I'd just remove this line.

marc-hanheide commented:
If the example is amended accordingly, this can be merged.

marc-hanheide commented:
Also, the example won't compile nor run at the moment. It should be enabled and tested.

pet1330 commented:
I've removed the 3D from the example. I've also removed the optional compile for that component to check it compiles. If I get a chance I'll implement a 3D example at some point, but I'll submit that as a separate pull request.

marc-hanheide commented:
OK, now I'm :+1: for this, @cdondrup please give your final okay and merge.

cdondrup commented:
Looks alright.
